File: src/main/java/main/constants/TrackSource.java
package main.constants;

public enum TrackSource {
    SPOTIFY,
    DEEZER,
    YOUTUBE,
    SOUNDCLOUD;

    public static TrackSource parse(String name) {
        switch (name.toLowerCase()) {
            case "youtube" -> {
                return YOUTUBE;
            }
            case "spotify" -> {
                return SPOTIFY;
            }
            case "deezer" -> {
                return DEEZER;
            }
            case "soundcloud" -> {
                return SOUNDCLOUD;
            }
            default -> throw new NullPointerException("There is no such source");
        }
    }
}

File: src/main/kotlin/main/constants/TrackSource.kt
package main.constants

import java.util.*


enum class TrackSource {
    SPOTIFY,
    DEEZER,
    YOUTUBE,
    SOUNDCLOUD,
    APPLE_MUSIC,
    RESUMED;

    companion object {
        fun parse(name: String): TrackSource {
            return when (name.lowercase(Locale.getDefault())) {
                "youtube" -> YOUTUBE
                "spotify" -> SPOTIFY
                "deezer" -> DEEZER
                "soundcloud" -> SOUNDCLOUD
                "applemusic", "apple_music" -> APPLE_MUSIC
                "resumed" -> RESUMED
                else -> throw NullPointerException("There is no such source")
            }
        }
    }
}
